lib/krikri/enrichments/dcmi_type_map.rb

```diff
@@ -0,0 +1,121 @@
+                    'illumination' => RDF::DCMITYPE.Image,
+                    'poster' => RDF::DCMITYPE.Image,
+                    'appliance' => RDF::DCMITYPE.Image,
+                    'tool' => RDF::DCMITYPE.Image,
+                    'electronic component' => RDF::DCMITYPE.Image,
+                    'postcard' => RDF::DCMITYPE.Image,
+                    'equipment' => RDF::DCMITYPE.Image,
+                    'cartographic' => RDF::DCMITYPE.Image,
+                    'notated music' => RDF::DCMITYPE.Image,
+                    'mixed material' => RDF::DCMITYPE.Image,
+                    'text' => RDF::DCMITYPE.Text,
+                    'book' => RDF::DCMITYPE.Text,
+                    'publication' => RDF::DCMITYPE.Text,
+                    'magazine' => RDF::DCMITYPE.Text,
+                    'journal' => RDF::DCMITYPE.Text,
+                    'correspondence' => RDF::DCMITYPE.Text,
+                    'writing' => RDF::DCMITYPE.Text,
+                    'written' => RDF::DCMITYPE.Text,
+                    'manuscript' => RDF::DCMITYPE.Text,
+                    'audio' => RDF::DCMITYPE.Sound,
+                    'sound' => RDF::DCMITYPE.Sound,
+                    'oral history recording' => RDF::DCMITYPE.Sound,
+                    'finding aid' => RDF::DCMITYPE.Collection,
+                    'online collection' => RDF::DCMITYPE.Collection,
+                    'electronic resource' => RDF::DCMITYPE.InteractiveResource,
+                    'video game' => RDF::DCMITYPE.InteractiveResource,
+                    'online exhibit' => RDF::DCMITYPE.InteractiveResource,
+                    'moving image' => RDF::DCMITYPE.MovingImage,
+                    'movingimage' => RDF::DCMITYPE.MovingImage,
+                    'motion picture' => RDF::DCMITYPE.MovingImage,
+                    'film' => RDF::DCMITYPE.MovingImage,
+                    'video' => RDF::DCMITYPE.MovingImage,
+                    'object' => RDF::DCMITYPE.PhysicalObject
+                  }
+
+    ##
+    # @param map [Hash<String, RDF::Vocabulary::Term>]
+    def initialize(map = nil)
+      @map = map || DEFAULT_MAP
+    end
+
+    ##
+    # @param value [Object] the value to enrich
+    #
+    # @return [DPLA::MAP::Controlled::DCMIType, nil] the matching DCMI Type
+    #   term. `nil` if no matches are found.
+    def enrich_value(value)
+      return value unless value.is_a? String
+
+      match = @map.fetch(value.downcase) { most_similar(value) }
+      return value if match.nil?
+      dcmi = DPLA::MAP::Controlled::DCMIType.new(match)
+      dcmi.prefLabel = match.label
+      dcmi
+    end
+
+
+    private
+
+    ##
+    # Performs White Similarity comparison against the keys, and gives the
+    # value of the closest match.
+    #
+    # @param value [String] a string value to compare to the hash map keys.
+    # @param threshold [Float] the value at which a string is considered to
+    #   be a match
+    #
```

Comment:
Yeah, and probably a link to a resource describing the algorithm and a `@todo` to evaluate text comparison methodologies/move text comparison logic to a utility.

:+1: